Cargo.toml

```diff
@@ -17,6 +17,9 @@ license = "MIT"
 [features]
 unstable = []
 
+[dependencies]
+slotmap = "1.0.6"
+
```

Comment:
`slotmap` is pretty nifty, and used by many packages (including Bevy i think!) but honestly this PR doesn't require a lot of its features.

all that the current Earley parser really uses it for is a "read only" arena allocator. This may be better served by something like [`elsa::FrozenVec`](https://docs.rs/elsa/latest/elsa/vec/struct.FrozenVec.html) which is an append-only vector.

Reply:
@shnewto i removed the usage of slotmap in the latest commit, instead using a minimal append-only wrapper around `Vec`.

this didn't have much impact on performance, but maybe worth the extra code to remove `slotmap` ?